src/index.ts

```diff
@@ -282,92 +286,101 @@ class CruxPayPeer extends EventEmitter {
-        await (this._payIDClaim as PayIDClaim).encrypt();
+        try {
+            const clientMapping: any = this._clientMapping;
+            const csAddressMap: any = {};
+            for (const key of Object.keys(newAddressMap)) {
+                csAddressMap[clientMapping[key]] = newAddressMap[key];
+            }
 
-        if (!acknowledgement) { throw new Error((`Could not update the addressMap`)); }
-        return acknowledgement;
+            await (this._payIDClaim as PayIDClaim).decrypt();
+            const acknowledgement = await (this._nameservice as nameservice.NameService).putAddressMapping({secrets: (this._payIDClaim as PayIDClaim).identitySecrets}, csAddressMap);
+            await (this._payIDClaim as PayIDClaim).encrypt();
+            return acknowledgement;
+        } catch (error) {
+            throw error.CruxClientError.fromError(error);
+        }
     }
 
     public getAddressMap = async (): Promise<IAddressMapping> => {
-        const clientMapping: any = this._clientMapping;
-        const clientIdToAssetIdMap: any = {};
-        for (const i of Object.keys(clientMapping)) {
-            clientIdToAssetIdMap[clientMapping[i]] = i;
-        }
+        try {
+            const clientMapping: any = this._clientMapping;
+            const clientIdToAssetIdMap: any = {};
+            for (const i of Object.keys(clientMapping)) {
+                clientIdToAssetIdMap[clientMapping[i]] = i;
+            }
 
-        const clientIdMap: any = {};
-        if (this._payIDClaim && this._payIDClaim.virtualAddress) {
-            const assetIdMap = await (this._nameservice as nameservice.NameService).getAddressMapping(this._payIDClaim.virtualAddress);
+            const clientIdMap: any = {};
+            if (this._payIDClaim && this._payIDClaim.virtualAddress) {
+                const assetIdMap = await (this._nameservice as nameservice.NameService).getAddressMapping(this._payIDClaim.virtualAddress);
 
-            for (const key of Object.keys(assetIdMap)) {
-                clientIdMap[clientIdToAssetIdMap[key]] = assetIdMap[key];
-            }
-            return clientIdMap;
+                for (const key of Object.keys(assetIdMap)) {
+                    clientIdMap[clientIdToAssetIdMap[key]] = assetIdMap[key];
+                }
+                return clientIdMap;
 
-        } else {
-            return {};
+            } else {
+                return {};
+            }
+        } catch (error) {
+            throw error.CruxClientError.fromError(error);
         }
     }
 
+    private getIDStatus = async (): Promise<nameservice.CruxIDRegistrationStatus> => {
+        await (this._payIDClaim as PayIDClaim).decrypt();
+        const result = (this._nameservice as nameservice.NameService).getRegistrationStatus({secrets: (this._payIDClaim as PayIDClaim).identitySecrets});
+        await (this._payIDClaim as PayIDClaim).encrypt();
+        return result;
+    }
+
 }
 
 export {
-    Errors,
```

Comment:
we should be exposing the new `error` scope outside for wallet clients to catch with `instanceof` or `packageErrorCode` if required.